marklogic-data-hub-central/src/main/java/com/marklogic/hub/central/managers/EntitySearchManager.java

```diff
@@ -190,33 +189,35 @@ private StructuredQueryDefinition buildQuery(QueryManager queryMgr, SearchQuery
         searchQuery.getQuery().getSelectedFacets().forEach((facetType, data) -> {
             StructuredQueryDefinition facetDef = null;
 
-            if (facetType.equals(COLLECTION_CONSTRAINT_NAME)) {
-                facetDef = queryBuilder
-                    .collectionConstraint(facetType, data.getStringValues().toArray(new String[0]));
-            }
-            else if (facetType.equals(JOB_RANGE_CONSTRAINT_NAME)) {
-                facetDef = queryBuilder
-                    .wordConstraint(JOB_WORD_CONSTRAINT_NAME,
-                        data.getStringValues().toArray(new String[0]));
-            }
-            else if (facetType.equals(CREATED_ON_CONSTRAINT_NAME)) {
-                // Converting the date in string format from yyyy-MM-dd format to yyyy-MM-dd HH:mm:ss format
-                LocalDate startDate = LocalDate.parse(data.getRangeValues().getLowerBound(), DATE_FORMAT);
-                String startDateTime = startDate.atStartOfDay(ZoneId.systemDefault())
-                    .format(DATE_TIME_FORMAT);
-
-                // Converting the date in string format from yyyy-MM-dd format to yyyy-MM-dd HH:mm:ss format
-                // Adding 1 day to end date to get docs harmonized on the end date as well.
-                LocalDate endDate = LocalDate.parse(data.getRangeValues().getUpperBound(), DATE_FORMAT)
-                    .plusDays(1);
-                String endDateTime = endDate.atStartOfDay(ZoneId.systemDefault()).format(DATE_TIME_FORMAT);
-
-                facetDef = queryBuilder
-                    .and(queryBuilder.rangeConstraint(facetType, Operator.GE, startDateTime),
-                        queryBuilder.rangeConstraint(facetType, Operator.LT, endDateTime));
-            }
-            else { // If a property is not a Hub property, then it is an Entity Property
-                facetDef = getEntityPropertyConstraints(facetType, data, queryBuilder);
+            switch (facetType) {
```

Comment:
Something to consider as this grows - a switch statement is often a bad practice because it naturally creates a large block of code with explicit knowledge of the various choices for the value being switched on. Switch statements can usually be replaced by a more "open/closed" mechanism via an interface like this:

     public interface FacetHandler 
       StructuredQueryDefinition buildQuery(FacetData data, StructuredQueryBuilder queryBuilder)

This class then has a Map<String, FacetHandler>, where String is the constraint name. And for each selected facet, it finds the matching FacetHandler and asks it to build an SQD.

The benefit of this is that you can easily unit-test each implementation of FacetHandler in isolation. And you avoid an ungainly switch statement and have a much more extensible design - when it's time to support a new constraint, you just create a new instance of FacetHandler and add it to the map. 

It may be worth making that change now, particularly if there's not enough coverage of the code in each switch block here.